### PR TITLE
chore: upgrade dependencies / better health checks

### PR DESCRIPTION
**Pull request by tjwelde**
## fixes KILTProtocol/ticket#1082
This PR upgrades nest and adds health checks.

## How to test:
- Try out all endpoints
- Check out `/health` endpoint

## Checklist:

- [ ] I have verified that the code works
- [x] I have verified that the code is easy to understand
  - [ ] If not, I have left a well-balanced amount of inline comments
- [x] I have [left the code in a better state](https://deviq.com/boy-scout-rule/)
- [ ] I have documented the changes (where applicable)

### CODE REVIEW

**Comment by rflechtner** [comment]
I realised we could also check for a working blockchain connection and played around with that. Could add that here

File: src/health/health.controller.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    private health: HealthCheckService,
+    private dns: DNSHealthIndicator,
+    private mongoose: MongooseHealthIndicator
+  ) {}
+
+  @Get()
+  @HealthCheck()
+  check() {
+    return this.health.check([
+      () => this.dns.pingCheck('google', 'https://google.com'),
```
Comment (rflechtner) [inline src/health/health.controller.ts:21]: is this from playing around or do we need a health check that pings an external site?

Comment (tjwelde) [reply src/health/health.controller.ts:21]: This would just make sure, that the service can reach the internet.

Comment (rflechtner) [reply src/health/health.controller.ts:23]: should we ping example.com instead?

**Review by rflechtner — APPROVED** [review]
LGTM

**Pull request merged**